compiler/src/main/kotlin/ast/mapper.kt

```diff
@@ -109,34 +114,55 @@ private fun visitExpr(node: ParseTree): Expr {
             ctx = node,
-            ident = node.var_ident().text
+            ident = node.ident.text
         )
-        is CellmataParser.FuncExprContext -> throw NotImplementedError()
-
-        is CellmataParser.ExprContext -> visitExpr(node.expr_2())
-        is CellmataParser.Expr3ContContext -> visitExpr(node.expr_3())
-        is CellmataParser.Expr4ContContext -> visitExpr(node.expr_4())
-        is CellmataParser.Expr5ContContext -> visitExpr(node.expr_5())
-        is CellmataParser.Expr6ContContext -> visitExpr(node.expr_6())
-        is CellmataParser.Expr7ContContext -> visitExpr(node.expr_7())
-        is CellmataParser.Expr8ContContext -> visitExpr(node.expr_8())
-        is CellmataParser.Expr9ContContext -> visitExpr(node.expr_9())
-        is CellmataParser.Expr10ContContext -> visitExpr(node.expr_10())
-        is CellmataParser.Expr11ContContext -> visitExpr(node.expr_11())
-
-        else -> throw AssertionError()
+        is CellmataParser.FuncExprContext -> FuncExpr(
+            ctx = node.value,
+            args = node.value.expr().map { visitExpr(it) }
+        )
+        is CellmataParser.StateIndexExprContext -> StateIndexExpr(ctx = node)
+        is CellmataParser.ArrayValueExprContext -> ArrayBodyExpr(
+            ctx = node.array_value(),
+            values = node.array_value().array_body().expr().map(::visitExpr)
+        )
+        else -> throw AssertionError("Unexpected tree node")
     }
 }
 
 private fun visitStmt(node: ParseTree): Stmt {
-    return when(node) {
+    return when (node) {
         is CellmataParser.Assign_stmtContext -> AssignStmt(ctx = node, expr = visitExpr(node.expr()))
-        is CellmataParser.If_stmtContext -> throw NotImplementedError() //ToDo
+        is CellmataParser.If_stmtContext -> IfStmt(
+            ctx = node,
+            conditionals = listOf( // Create list of list of ConditionalBlocks, then flatten to list of ConditionalBlocks
+                listOf(
+                    ConditionalBlock( // If clause
+                        ctx = node,
+                        expr = visitExpr(node.if_stmt_if().if_stmt_block().if_stmt_condition().expr()),
+                        block = visitCodeBlock(node.if_stmt_if().if_stmt_block().code_block())
+                    )
+                ),
+                node.if_stmt_elif().map { // Elif clauses
+                    ConditionalBlock(
+                        ctx = node,
+                        expr = visitExpr(it.if_stmt_block().if_stmt_condition().expr()),
+                        block = visitCodeBlock(it.if_stmt_block().code_block())
+                    )
+                }
+            ).flatten(),
+            elseBlock = when (node.if_stmt_else()) {
+                // null???
+                is CellmataParser.If_stmt_elseContext -> visitCodeBlock(node.if_stmt_else().code_block())
+                else -> throw AssertionError("Unexpected tree node")
+            }
```

Comment:
Here will be problems when the else-case is not present as the comment `// nul???` also suggests